### ISSUE TITLE
data.key and international keyboards

### ISSUE THREAD

**Issue by amirmohsen** [ianstormtaylor/slate#958]
#### Do you want to request a *feature* or report a *bug*?

bug

#### What's the current behavior?

Your rich text example uses `data.key` to handle keyboard shortcuts [on this line](https://github.com/ianstormtaylor/slate/blob/master/examples/rich-text/index.js#L111).

You're using the backtick to handle code blocks. The problem is `data.key` uses codes that are specific to the US keyboard and don't necessarily work for other keyboards. The key code that you are using is 192 but the code for backtick on my keyboard (UK layout) is 223 which results in `data.key` being undefined.

#### What's the expected behavior?

To get more accurate key characters across all keyboards, you should use [`KeyboardEvent.key`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) instead.

**Comment by danburzo** [comment]
On a tangent, I'd venture to say the `data` object from the `onKeyDown` callback should be deprecated (and possibly reintroduced as a helper). I think the decision on how to interpret `KeyboardEvent`s should be the responsibility of the developer.

**Comment by ianstormtaylor** [comment]
@danburzo would love your thoughts in that related thread if you have any!

@amirmohsen I'm open to that change, and would love a PR!

**Comment by danburzo** [comment]
@ianstormtaylor I added a few thoughts there. 
As for the current conversation, the reason why I've been reluctant to change from `event.which` to the more recently introduced properties (`key` and `code`) is twofold:

1. not sure how to decide when to put keyboard shortcuts on "physical" keys vs. "actual" keys — "actual" keys is kind of frustrating on RO layout / US keyboard, as the right part of the keyboard, where a lot of coding punctuation happens, is taken up with diacritical marks.

2. also not sure about the extent of the support, although I'd think most major bugs across browsers have been fixed
